cpp/log/cert.cc

```diff
@@ -53,8 +53,17 @@ string ASN1ToStringAndCheckForNulls(ASN1_STRING* asn1_string,
   const string cpp_string(reinterpret_cast<char*>(
       ASN1_STRING_data(asn1_string)), ASN1_STRING_length(asn1_string));
 
+  // Unfortunately ASN1_STRING_length returns a signed value
+  if (ASN1_STRING_length(asn1_string) < 0) {
+    *status = util::Status(Code::INVALID_ARGUMENT, "ASN1 string is corrupt?");
+  }
+
+  // We now know it's not a negative length so either size_t is signed in which
```

Comment:
`size_t` is defined as an unsigned integer (there's a signed version too: `ssize_t`)